### ISSUE TITLE
Error trying to build psutil from source with mingwpy-0.1.0b3

### ISSUE THREAD

**Issue by ogrisel** [mingwpy/mingwpy.github.io#30]
Here is the build message I got trying to build the latest release tag `release-3.4.2` of the https://github.com/giampaolo/psutil repo:

```
building 'psutil._psutil_windows' extension
C:\msys64\home\ogrisel\venv\mingwpy\Scripts\gcc.exe -mdll -O -Wall -DPSUTIL_VERSION=342 -D_WIN32_WINNT=0x0602 -D_AVAIL_WINVER_=0x0602 -D_CRT_SECURE_NO_WARNINGS -DPSAPI_VERSION=1 -Ic:\python34\include -Ic:\python34\include -c psutil/_psutil_windows.c -o build\temp.win-amd64-3.4\Release\psutil\_psutil_windows.o
In file included from C:/msys64/home/ogrisel/venv/mingwpy/share/mingwpy/x86_64-w64-mingw32/include/iprtrmib.h:9:0,
                 from C:/msys64/home/ogrisel/venv/mingwpy/share/mingwpy/x86_64-w64-mingw32/include/iphlpapi.h:17,
                 from psutil/_psutil_windows.c:21:
C:/msys64/home/ogrisel/venv/mingwpy/share/mingwpy/x86_64-w64-mingw32/include/mprapi.h:865:3: error: unknown type name 'CERT_NAME_BLOB'
   CERT_NAME_BLOB *certificateNames;
   ^
C:/msys64/home/ogrisel/venv/mingwpy/share/mingwpy/x86_64-w64-mingw32/include/mprapi.h:887:3: error: unknown type name 'CRYPT_HASH_BLOB'
   CRYPT_HASH_BLOB certBlob;
   ^
psutil/_psutil_windows.c:26:0: warning: ignoring #pragma comment  [-Wunknown-pragmas]
 #pragma comment(lib, "IPHLPAPI.lib")
 ^
In file included from psutil/arch/windows/process_info.h:13:0,
                 from psutil/_psutil_windows.c:31:
psutil/arch/windows/ntextapi.h:23:14: error: nested redefinition of 'enum _KWAIT_REASON'
 typedef enum _KWAIT_REASON {
              ^
psutil/arch/windows/ntextapi.h:23:14: error: redeclaration of 'enum _KWAIT_REASON'
In file included from psutil/arch/windows/ntextapi.h:8:0,
                 from psutil/arch/windows/process_info.h:13,
                 from psutil/_psutil_windows.c:31:
C:/msys64/home/ogrisel/venv/mingwpy/share/mingwpy/x86_64-w64-mingw32/include/winternl.h:622:16: note: originally defined here
   typedef enum _KWAIT_REASON {
                ^
...
```

There are many more errors after that. Here is the full log: https://gist.githubusercontent.com/ogrisel/e08d59f9c8ac500855bc/raw/6c539599c759583508f6d1b2c3f9e7e4395630da/psutil-build.log

Environment:
- Windows 10 (64 bit)
- Python 3.4.4 or Python 2.7.11 (64 bit)

Note that the problem was already present in 0.1.0b1.

**Comment by ogrisel** [comment]
BTW @carlkl I think you should create the mingwpy main repo ASAP so that we can put such bug reports on the mingwpy issue tracker instead of the mingwpy.github.io issue tracker that should be dedicated to the website and the documentation of the project instead.

**Comment by ogrisel** [comment]
I can also reproduce those errors with my version of mingwpy-dev's gcc compiled from source.

**Comment by carlkl** [comment]
@ogrisel, is it https://github.com/giampaolo/psutil/issues/345#issuecomment-44022399?

**Comment by carlkl** [comment]
@ogrisel, I still get build errors with my local mingw-builds variant I have to figure out. This may take some more time as expected due to the long build times and the number combinations of setings I have to try out. Nevertheless I will create an empty repo for the wheel-builder.

**Comment by carlkl** [comment]
done: https://github.com/mingwpy/mingwpy-wheel-builder

**Comment by matthew-brett** [comment]
The original patch to get psutils working on mingw is attached to https://github.com/giampaolo/psutil/issues/158

The main author of psutil has also been planning to remove mingw32 code, on the basis it isn't working any more, and the code is messy, so it might be good to get this working soon: https://github.com/giampaolo/psutil/issues/591

**Comment by ogrisel** [comment]
Ok indeed. psutil is not a good test project for mingwpy then.

FYI I built scikit-learn and all tests pass. Cython seems clean too although I did not run the test suite yet. pandas is almost clean.

It could be interesting to try to build libxml2 / libxslt and lxml with mingwpy but that sounds more complicated. And maybe database connectors for postgresql and mysql, Pillow and pyzmq.

But the real deal is numpy & scipy + openblas  as AFAIK those are the only projects that really need gfortran :)
